src/tools/miri/src/shims/intrinsics/mod.rs

```diff
@@ -131,6 +131,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.write_pointer(Pointer::new(ptr.provenance, masked_addr), dest)?;
             }
 
+            // The default implementation always returns `false`, but we want
+            // to return either `true` or `false` at random.
+            "is_comptile_time_known" => {
+                let rand = 0;
+                _ = getrandom::getrandom(&mut [rand]);
+
+                this.write_scalar(
+                    Scalar::from_bool(if (rand % 1) == 1 { true } else { false }),
+                    dest,
+                )?;
+            }
```

Comment:
As I said before, please don't use `getrandom`. It's absolutely crucial that execution is deterministic. That's why I said

> Please just use this.machine.rng.get_mut().gen(), which can return a bool.

You marked that as resolved without actually resolving the issue. Please don't do that, it makes reviewing unnecessarily cumbersome.

Reply:
It's been changed again locally, yeah I probably shouldn't resolve before committing though

Reply:
There was no code push since your comment. This was likely resolved locally and not pushed yet